src/components/two-factor-authentication/TokenReset.vue

```diff
@@ -0,0 +1,190 @@
+<script setup lang="ts">
+  import { ref, type Ref } from 'vue';
+  import { useDisplay } from 'vuetify';
+  import LayoutCard from '@/components/cards/LayoutCard.vue';
+  import SpshTooltip from '@/components/admin/SpshTooltip.vue';
+  import { useI18n, type Composer } from 'vue-i18n';
+  import {
+    useTwoFactorAuthentificationStore,
+    type TwoFactorAuthentificationStore,
+  } from '@/stores/TwoFactorAuthentificationStore';
+  import type { Personendatensatz } from '@/stores/PersonStore';
+  import axios from 'axios';
+
+  const { t }: Composer = useI18n({ useScope: 'global' });
+  const { mdAndDown }: { mdAndDown: Ref<boolean> } = useDisplay();
+  const isTokenResetRequested: Ref<boolean> = ref(false);
+  const errorThrown: Ref<boolean> = ref(false);
+  const twoFactorAuthenticationStore: TwoFactorAuthentificationStore = useTwoFactorAuthentificationStore();
+  const dialogText: Ref<string> = ref('');
+  const dialogHeader: Ref<string> = ref(t('admin.person.twoFactorAuthentication.tokenReset'));
+
+  type Emits = {
+    (event: 'dialogClosed'): void;
+  };
+
+  const emits: Emits = defineEmits<{
+    (event: 'dialogClosed'): void;
+  }>();
+
+  type Props = {
+    errorCode: string;
+    disabled: boolean;
+    person: Personendatensatz;
+    tokenType: string | null;
+  };
+
+  const props: Props = defineProps<Props>();
+  async function closeTokenResetDialog(isActive: Ref<boolean>): Promise<void> {
+    isActive.value = false;
+    errorThrown.value = false;
+    isTokenResetRequested.value = false;
+    emits('dialogClosed');
+  }
+
+  function handleApiError(error: unknown): void {
+    errorThrown.value = true;
+    if (axios.isAxiosError(error)) {
+      if (error.response && error.response.data.i18nKey) {
+        const message: string =
+          t('admin.person.twoFactorAuthentication.errors.' + error.response.data.i18nKey) ||
+          t('admin.person.twoFactorAuthentication.errors.UNKNOWN_ERROR');
+        dialogText.value = message;
+        dialogHeader.value =
+          props.tokenType === 'hardware'
```

Comment:
Hier am besten ein Enum für "software" und "hardware" erstellen. Ich glaube den gibt es schon irgendwo.

Reply:
done. Ne, waren 2 Tickets an denen gleichzeitig gearbeitet wurde. Das hier ist in der TokenReset vue und war noch nicht drin